source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/actions/task_space_actions.py

```diff
@@ -192,3 +196,388 @@ def _compute_frame_jacobian(self):
+    def raw_actions(self) -> torch.Tensor:
+        return self._raw_actions
+
+    @property
+    def processed_actions(self) -> torch.Tensor:
+        return self._processed_actions
+
+    """
+    Operations.
+    """
+
+    def process_actions(self, actions: torch.Tensor):
+
+        # Update ee pose, which would be used by relative targets (i.e., pose_rel)
+        self._compute_ee_pose()
+
+        # Update task frame pose w.r.t. the root frame.
+        self._compute_task_frame_pose()
+
+        # Pre-process the raw actions for operational space control.
+        self._preprocess_actions(actions)
+
+        # set command into controller
+        self._osc.set_command(
+            command=self._processed_actions,
+            current_ee_pose_b=self._ee_pose_b,
+            current_task_frame_pose_b=self._task_frame_pose_b,
+        )
+
+    def apply_actions(self):
+
+        # Update the relevant states and dynamical quantities
+        self._compute_dynamic_quantities()
+        self._compute_ee_jacobian()
+        self._compute_ee_pose()
+        self._compute_ee_velocity()
+        self._compute_ee_force()
+        # Calculate the joint efforts
+        self._joint_efforts[:] = self._osc.compute(
+            jacobian_b=self._jacobian_b,
+            current_ee_pose_b=self._ee_pose_b,
+            current_ee_vel_b=self._ee_vel_b,
+            current_ee_force_b=self._ee_force_b,
+            mass_matrix=self._mass_matrix,
+            gravity=self._gravity,
+        )
+        self._asset.set_joint_effort_target(self._joint_efforts, joint_ids=self._joint_ids)
+
+    def reset(self, env_ids: Sequence[int] | None = None) -> None:
+        self._raw_actions[env_ids] = 0.0
+        if self._contact_sensor is not None:
+            self._contact_sensor.reset(env_ids)
+        if self._task_frame_transformer is not None:
+            self._task_frame_transformer.reset(env_ids)
+
+    """
+    Helper functions.
+
+    """
+
+    def _first_RigidObject_child_path(self):
+        """Finds the first RigidObject child under the articulation asset."""
+        child_prims = find_matching_prims(self._asset.cfg.prim_path + "/.*")
+        rigid_child_prim = None
+        # Loop through the list and stop at the first RigidObject found
+        for prim in child_prims:
+            if prim.HasAPI(UsdPhysics.RigidBodyAPI):
+                rigid_child_prim = prim
+                break
+        if rigid_child_prim is None:
+            raise ValueError("No child rigid body found under the expression: '{self._asset.cfg.prim_path}'/.")
+        rigid_child_prim_path = rigid_child_prim.GetPath().pathString
+        # Remove the specific env index from the path string
+        rigid_child_prim_path = self._asset.cfg.prim_path + "/" + rigid_child_prim_path.split("/")[-1]
+        return rigid_child_prim_path
+
+    def _resolve_command_indexes(self):
+        """Resolves the indexes for the various command elements within the command tensor."""
```

Comment:
please add google style docstrings for the exceptions raised

Reply:
def _resolve_command_indexes(self):
        """Resolves the indexes for the various command elements within the command tensor.

        Raises:
            ValueError: If any command index is left unresolved.
        """